From 6186992c69e04d555ba108217fa4d9f7bed075d1 Mon Sep 17 00:00:00 2001
From: Chad Reese <kreese@caviumnetworks.com>
Date: Thu, 2 Dec 2010 07:26:56 -0800
Subject: [PATCH 048/238] Octeon: RapidIO: Debug output modifications

Source: Cavium SDK 2.1.0-407

Minor improvements on debug output for soft packet fifo receive.

Signed-off-by: Chad Reese <kreese@caviumnetworks.com>
Integrated-by: Phil Staub <Phil.Staub@windriver.com>
---
 arch/mips/cavium-octeon/octeon-rapidio.c |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/octeon-rapidio.c b/arch/mips/cavium-octeon/octeon-rapidio.c
index e5ff387..c1f7d8d 100644
--- a/arch/mips/cavium-octeon/octeon-rapidio.c
+++ b/arch/mips/cavium-octeon/octeon-rapidio.c
@@ -534,7 +534,6 @@ static void octeon_rio_rx_soft_fifo(struct rio_mport *mport)
 		int tt;
 		/* Update the SKB to match the length of data in it */
 		__skb_put(skb, length);
-		DEBUG_PACKET(mport, "RX", "%d byte packet\n", skb->len);
 		/* Figure out if this packet is a port write. The contents
 		    of the skb is the raw SRIO packet without the first ackID
 		    and CRF byte */
@@ -545,9 +544,14 @@ static void octeon_rio_rx_soft_fifo(struct rio_mport *mport)
 		transaction = skb->data[(tt) ? 5 : 3] >> 4;
 		/* Call the port write handler if this is a port write */
 		if ((ftype == 0x8) && (transaction == 0x4)) {
+			DEBUG_PACKET(mport, "RX", "%d byte port write\n",
+				skb->len);
 			rio_inb_pwrite_handler((union rio_pw_msg *)
 				(skb->data + ((tt) ? 11 : 9)));
 		}
+		else
+			DEBUG_PACKET(mport, "RX", "%d byte unknown packet\n",
+				skb->len);
 	}
 	dev_kfree_skb_any(skb);
 }
@@ -764,8 +768,8 @@ static int __init octeon_rio_init(void)
 	}
 
 	/* The links sometimes take a little bit to come up. Delay a half
-	   second. Linux currently doesn't ever try to enumerate if the
-	   links come up after this call */
+	    second. Linux currently doesn't ever try to enumerate if the
+	    links come up after this call */
 	msleep(500);
 	return rio_init_mports();
 }
-- 
1.7.0

